pkgs/tools/admin/pgadmin/default.nix

```diff
@@ -0,0 +1,152 @@
+    packageJSON = ./package.json;
+    yarnLock = ./yarn.lock;
+    yarnNix = ./yarn.nix;
+  };
+in
+
+python3Packages.buildPythonApplication rec {
+  inherit pname version src;
+
+  # from Dockerfile
+  CPPFLAGS = "-DPNG_ARM_NEON_OPT=0";
+
+  format = "other";
+
+  postPatch = ''
+    # don't build sphinx documentation
+    sed -r 's|LC_ALL=en_US.UTF-8.* Makefile.sphinx .*$|true|g' -i Makefile
+    patchShebangs .
+    # relax dependencies
+    substituteInPlace requirements.txt \
+    --replace "Pillow==8.3.*" "Pillow>=8.3.0" \
+    --replace "psycopg2==2.8.*" "psycopg2>=2.8.0" \
+    --replace "cryptography==3.*" "cryptography>=3.0" \
+    --replace "requests==2.25.*" "requests>=2.25.0"
+    # don't use Server Mode (can be overridden later)
+    substituteInPlace pkg/pip/setup_pip.py \
+    --replace "req = req.replace('psycopg2', 'psycopg2-binary')" "req = req" \
+    --replace "builtins.SERVER_MODE = None" "builtins.SERVER_MODE = False"
+  '';
+
+  buildPhase = ''
+    # Adapted from pkg/pip/build.sh
+    echo Creating required directories...
+    mkdir pip-build
+
+    mkdir pip-build/pgadmin4
+    mkdir pip-build/pgadmin4/docs
+
+    # Build the clean tree
+    cd web
+    cp -r * ../pip-build/pgadmin4
+    cd ../docs
+    cp -r * ../pip-build/pgadmin4/docs
+    cd ../
+
+    cp -r ${yarnDeps}/* pip-build/pgadmin4
+
+    echo Creating distro config...
+    echo HELP_PATH = \'../../docs/en_US/_build/html/\' > pip-build/pgadmin4/config_distro.py
+    echo MINIFY_HTML = False >> pip-build/pgadmin4/config_distro.py
+
+    echo Creating manifest...
+    echo recursive-include pgadmin4 \* > pip-build/MANIFEST.in
+
+    echo Building wheel...
+    cd pip-build
+    ${python3}/bin/python ../pkg/pip/setup_pip.py bdist_wheel
+    cd ../
+
+    # Get the build
+    if [ ! -d dist ]; then
+        mkdir dist
+    fi
+    cp pip-build/dist/*.whl dist/
+  '';
+
+  installPhase = ''
+    ${python3}/bin/python -m pip install dist/*.whl --prefix $out
+  '';
+
+  nativeBuildInputs = [ python3Packages.cython python3Packages.pip ];
+  buildInputs = [
+    zlib
+    python3Packages.wheel
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    flask
+    flask-gravatar
+    flask_login
+    flask_mail
+    flask_migrate
+    flask_sqlalchemy
+    flask_wtf
+    flask-compress
+    passlib
+    pytz
+    simplejson
+    six
+    speaklater3
+    sqlparse
+    wtforms
+    flask-paranoid
+    psutil
+    psycopg2
+    python-dateutil
+    sqlalchemy
+    itsdangerous
+    flask-security-too
+    bcrypt
+    cryptography
+    sshtunnel
+    ldap3
+    flask-babelex
+    flask-babel
+    gssapi
+    flask-socketio
+    eventlet
+    httpagentparser
+    user-agents
+    wheel
+    authlib
+    qrcode
+    pillow
+    pyotp
+  ];
+
+  # pgadmin offers a wide range of tests in web/regression
+  # unfortunately most of them require a PostgresServer in the
+  # test environment. Also, the data and log directories need
+  # to exist in the test environment.
```

Comment:
This can be done in a VM test. We should probably do that to make sure dependency updates do not cause regressions.

Reply:
How would you go about it?

Reply:
I didn't write a test yet but I would go into https://github.com/NixOS/nixpkgs/tree/master/nixos/tests and look at other ones and try to find something similar to my use case.

Reply:
Hi @SuperSandro2000 
I added the test suite which runs all python related tests. I'd love to have your feedback on them :+1: